src/iexamine.cpp

```diff
@@ -3594,105 +3595,153 @@ void iexamine::climb_down( player &p, const tripoint &examp )
+                            p.home.set_ammo_ration(lv - 1);
+                        }
+                        case 4:{ //meds
+                            int lv = menu(true, _("Medicine Rationing Level:"),
+                                _("None: Use/take anything and everything you want."),
+                                _("Light: Use medicine to treat apropriate injuries/illness only."),//no recreational use and use a bandage not a first aid kit for that scratch.
+                                _("Moderate: \"It's just a flesh wound.\"  Tough out minor wounds."),
+                                _("Heavy: \"Tis But a scratch!\"  Ignore all but the most serious wounds."), NULL);//switched quotation with med. For MP refrence.
+                            p.home.set_med_ration(lv - 1);
+                        }
+                    } //ratmenu switch
+                } while (ratmenu.ret != 0);//loop rationing options
+            } //desktop case 1 rationing
+            case 2:{ //ammo type prefrences
+                add_msg("Not implemented yet.\n(╮°-°)╮┳━━┳ ( ╯°□°)╯ ┻━━┻");
+                //TODO: Menu assign prefrence (weight) to ammo types.  Prefrence to be refrenced by NPC choosing weapon
+                //Use relayering UI?
+            }
+            case 3:{ //crafting priorities
+                add_msg("Not implemented yet.\n(╮°-°)╮┳━━┳ ( ╯°□°)╯ ┻━━┻");
+                //TODO: Menu assign priorities (weight) to craft recipies.  NPCs to refrence priorities when crafting.
+            }
+            case 4:{ //view status.
+                if (base.get_level == 1) iexamine::base_view(p, examp);
+                else iexamine::base_view_ele(p, examp);
+            }
+            case 5:{ //base planner
+                add_msg("Not implemented yet.\n(╮°-°)╮┳━━┳ ( ╯°□°)╯ ┻━━┻");
+            }
+            case 6:{ //personel managment
+                add_msg("Not implemented yet.\n(╮°-°)╮┳━━┳ ( ╯°□°)╯ ┻━━┻");
+            }
+        } //desktop switch
+        desktop.query();
+    } while (desktop.ret); //desktop loop
 }
 
 /**
  * View base status/information from a bulletin board. (limited info)
+ * 
+ * This is supposed to represent viewing stuff posted on a bulletin board so,
+ * displaying the info in a whole bunch of small frames like they are diffrent notes should
+ * be a future goal.
  */
 void iexamine::base_view( player &p, const tripoint &examp) {
-    add_msg("Not implemented yet.\n(╮°-°)╮┳━━┳ ( ╯°□°)╯ ┻━━┻");
+    auto &base = p->home;
+    //Window size values are currently arbitrary but window shouldn't be full screen if possible
+    //This function is in the source but no header refrence...
+    //static WINDOW_PTR board = output::create_popup_window( 40, 40, output::PopupFlags::PF_GET_KEY);
+    
+    //Simple implementation for now
+    std::list<std::string> basicInfo;
+    basicInfo.push_back( string_format( _("%s's Home Sweet Base%n"), p.get_name(), int len );
```

Comment:
`int len`?

Reply:
I was going to use it to create the same underline that the other sections had. %n stores the number of characters written.

Reply:
Sure, but I was referring to the invalid syntax. And you don't need this. You can just examine the formatted string itself. And you don't actually care about the numbers of *characters*, you care about the display width (see `utf8_width`). There are for example Unicode combination marks that take up *no* additional space on the console.

Reply:
thanks, Learned something new.